core/src/main/java/org/kiwix/kiwixmobile/core/utils/SharedPreferenceUtil.kt

```diff
@@ -131,15 +129,18 @@ class SharedPreferenceUtil @Inject constructor(context: Context?) {
 
   val nightMode: NightModeConfig.Mode
     get() = from(
-      sharedPreferences.getString(PREF_NIGHT_MODE, "" + AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
+      sharedPreferences.getString(
+        PREF_NIGHT_MODE,
+        "" + AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
+      )!!
         .toInt()
     )
```

Comment:
Addition of `!!` was the only method(to me) to eliminate the warning being showed here. If there is something better than this or previous implementations, please tell

Reply:
what we could do is for the default value return null and elvis operator to `AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM` before calling `toInt`

Reply:
I tried this, but couldn't express it in code. What i did was: 
` get() = null ?: from(
      sharedPreferences.getString(PREF_NIGHT_MODE, "" + AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM).toInt()`

How i was parsing it is: that it'll return null when the input is null and the `MODE_NIGHT_FOLLOW_SYSTEM.toInt()`
otherwise.
Can you please help me pointing out error

Reply:
when I refered to default value I meant the `getString` method as the second paramter is called `defaultValue`

`getString(key, null)?.toInt() ?: AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM `

a null on the LHS of a `?:` will just always execute the RHS

Reply:
Oh let it be, i'll try learning it from the koans that you mentioned above and if still unable to figure it out then will ask you again.
Please grant some time for that.

Reply:
Apologies, your comment wasn't being displayed while i was writing the trailing one. 
Thanks for helping!